labrad/wrappers.py

```diff
@@ -389,18 +389,21 @@ def _send(self, *args, **kw):
 
 @inlineCallbacks
 def getConnection(host=C.MANAGER_HOST, port=None, name="Python Client",
-                  password=None, tls_mode=C.MANAGER_TLS):
+                  password=None, tls_mode=C.MANAGER_TLS, username=None,
+                  headless=False):
     """Connect to LabRAD and return a deferred that fires the protocol object."""
     p = yield protocol.connect(host, port, tls_mode)
-    yield p.authenticate(password)
+    yield p.authenticate(password, username, headless)
     yield p.loginClient(name)
     returnValue(p)
 
 @inlineCallbacks
 def connectAsync(host=C.MANAGER_HOST, port=None, name="Python Client",
-                 password=None, tls_mode=C.MANAGER_TLS):
+                 password=None, tls_mode=C.MANAGER_TLS, username=None,
+                 headless=False):
     """Connect to LabRAD and return a deferred that fires the client object."""
-    p = yield getConnection(host, port, name, password, tls_mode=tls_mode)
+    p = yield getConnection(host, port, name, password, tls_mode=tls_mode,
+                            username=username, headless=headless)
```

Comment:
I agree. I was thinking about collecting these all into some sort of "ConnectOptions" class that would get passed around. Only the top-level "labrad.connect" function would accept separate keyword args. Maybe it's worth making that change here.